### ISSUE TITLE
Tag status: Improve clarity in screen reader text

### ISSUE THREAD

**Issue by jeana-adhoc** [department-of-veterans-affairs/vets-design-system-documentation#5011]
## Description

In the new Tag Status component, we are making SR-only text baked in, to announce "Status" for screen reader users. However, this does not provide the same context that sighted users are getting through color and icon use. The screen reader user needs to get the same meaning of "error", "success", etc that sighted users get. 

We should: 
- Update the hard-coded "Status" text in the sr-only element to include information about the related status. For example, "Status {statusType}: {tagText}", which might render something like "Status error: Missing information"
- Allow teams to customize the "Status {tagType}" text through a prop. This would give teams flexibility to customize the prefix text in the event that their tag text generates a confusing readout when combined with our status prefix (For example, "Status error: Email error"), or if they need to account for translations.

**Comment by amyleadem** [comment]
Update:
Oct 30, 2025: I changed the description of this issue to match the accessibility audit findings. This changes it from an enhancement request to a bug. @jeana-adhoc let me know if you want any changes here.
